pkg/controllers/webhookconfig/webhookconfig.go

```diff
@@ -126,9 +133,19 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options)
 }
 
 func (r *Reconciler) ReadyCheck(_ *http.Request) error {
-	r.rdyMu.Lock()
-	defer r.rdyMu.Unlock()
-	if !r.ready {
+	// skip readiness check if we're not leader
+	// as we depend on caches and being able to reconcile Webhooks
+	if !r.leaderElected {
+		select {
+		case <-r.leaderChan:
+			r.leaderElected = true
+		default:
+			return nil
+		}
+	}
+	r.webhookReadyMu.Lock()
```

Comment:
Move this lock above the `!r.leaderElected` check. :)

Reply:
why though, we're not accessing `webhookReady` :thinking: ? Just to protect the whole `func` ?

Reply:
Well, technically we don't, but the tests do. :/ You are using it in an eventually and then in another test or something like that. It's a bit weird. Normally, I think it should be fine though.

Reply:
Oh wait... I think the new configuration made it work. You slightly changed something in the end point thingy.